nomt/src/beatree/ops/mod.rs

```diff
@@ -59,24 +60,13 @@ pub fn lookup(
 /// node pointer whose separator is less than or equal to the given key.
 fn search_branch(branch: &branch::BranchNode, key: Key) -> Option<(usize, PageNumber)> {
     let prefix = branch.prefix();
-
-    match key.view_bits::<Msb0>()[..prefix.len()].cmp(prefix) {
-        Ordering::Equal => {}
-        Ordering::Less => return None,
-        Ordering::Greater => {
-            let i = branch.n() as usize - 1;
-            return Some((i, branch.node_pointer(i).into()));
-        }
-    }
-
-    let post_key = &key.view_bits::<Msb0>()[prefix.len()..];
-
     let mut low = 0;
     let mut high = branch.n() as usize;
 
     while low < high {
         let mid = low + (high - low) / 2;
-        if post_key < branch.separator(mid) {
+        let s = reconstruct_key(prefix, branch.separator(mid));
```

Comment:
remember: we _don't_ want to pessimize the typical case. this pessimizes the typical case by doing more work.